### ISSUE TITLE
BasicPubSub crash

### ISSUE THREAD

**Issue by jonyt** [aws/aws-iot-device-sdk-java-v2#188]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x ] I've searched for [previous similar issues](https://github.com/aws/aws-iot-device-sdk-java-v2/issues) and didn't find any solution

**Known Issue**
- [x ] I'm using ATS data type endpoint: the endpoint should look like `<prefix>-ats.iot.<region>.amazonaws.com`

**Describe the bug**
I'm running the BasicPubSub example with one modification - I'm adding a lastWill message. So I'm inserting the line `.withWill(new MqttMessage("topic", "".getBytes(java.nio.charset.StandardCharsets.UTF_8)), QualityOfService.AT_LEAST_ONCE, true)` [here](https://github.com/aws/aws-iot-device-sdk-java-v2/blob/fd6f9f36cfa012f72444cdb55b59c2a65f8a3e27/samples/BasicPubSub/src/main/java/pubsub/PubSub.java#L242). The run command is `java  -jar target\BasicPubSub-1.0-SNAPSHOT-jar-with-dependencies.jar --clientId the-thing  --x509rootca root-CA.crt --x509cert cert.pem --x509key cert.private_key --endpoint xxxxxxxxxxx-ats.iot.eu-west-1.amazonaws.com --count 5 -w --region eu-west-1 --x509 --x509rolealias iot-device-authorization-role-alias --x509endpoint xxxxxxxx.credentials.iot.eu-west-1.amazonaws.com --x509thing the-thing -m yoyo -t fff/the-thing`. 

The error I'm getting is below. When I add the CRT log there are no additional errors. Note that in the error there's a hardcoded path to the file `C:\Program Files (x86)\Jenkins\workspace\aws-crt-java-build-dll-win64\aws-crt-java\crt\aws-c-mqtt\source\packets.c:`, which doesn't appear on my computer. This may be related to issues #53 and #123, as the errors appear to be similar.

**SDK version number**
1.2.4 but also tested with latest 1.5.3

**Platform/OS/Hardware/Device**
Windows 11.

**Expected behavior**
I expect the program to send and receive messages normally. Without the line I added it works without issue.

**Logs/output**
The output I'm getting is:

```
Fatal error condition occurred in C:\Program Files (x86)\Jenkins\workspace\aws-crt-java-build-dll-win64\aws-crt-java\crt\aws-c-mqtt\source\packets.c:23: cur.ptr && cur.len
Exiting Application
at 0x00007FFD9DB1D062: Failed to lookup symbol: error 126
at 0x00007FFD9DB1D2CF: Failed to lookup symbol: error 126
at 0x00007FFD9DB0FEF7: Failed to lookup symbol: error 126
at 0x00007FFD9DA9768F: Failed to lookup symbol: error 126
at 0x00007FFD9DA95E55: Failed to lookup symbol: error 126
at 0x00007FFD9DA9146B: Failed to lookup symbol: error 126
at 0x00007FFD9DA9218F: Failed to lookup symbol: error 126
at 0x00007FFD9DB32C9F: Failed to lookup symbol: error 126
at 0x00007FFD9DA6D341: Failed to lookup symbol: error 126
at 0x00007FFD9DA6C935: Failed to lookup symbol: error 126
at 0x00007FFD9DA7DD64: Failed to lookup symbol: error 126
at 0x00007FFD9DA7F315: Failed to lookup symbol: error 126
at 0x00007FFD9DA7D4F7: Failed to lookup symbol: error 126
at 0x00007FFD9DA6E696: Failed to lookup symbol: error 126
at 0x00007FFD9DA6AC1D: Failed to lookup symbol: error 126
at 0x00007FFD9DA698FA: Failed to lookup symbol: error 126
at 0x00007FFD9DAF3265: Failed to lookup symbol: error 126
at 0x00007FFD9DAEE684: Failed to lookup symbol: error 126
at 0x00007FFD9DB00A6A: Failed to lookup symbol: error 126
at 0x00007FFD9DB00D25: Failed to lookup symbol: error 126
at 0x00007FFD9DAFECB4: Failed to lookup symbol: error 126
at 0x00007FFD9DB030A5: Failed to lookup symbol: error 126
at 0x00007FFD9DB1CF52: Failed to lookup symbol: error 126
at 0x7FFE164A7020: BaseThreadInitThunk
at 0x7FFE16C02630: RtlUserThreadStart
```

**Comment by bretambrose** [comment]
Looks like the packet encoding doesn't handle empty will payloads properly.  There's a related issue with empty payloads on received messages (https://github.com/awslabs/aws-c-mqtt/issues/204).  I'll try and slip in fixes for both today if possible.

**Comment by bretambrose** [comment]
This should be fixed in v1.5.4
